tests/testthat/test-extract_hash_from_object.R

```diff
@@ -0,0 +1,65 @@
+test_that("extract_hash_from_object string objects work", {
+  # test strings
+  expect_identical(extract_hash_from_object("foo"), "bd40ef")
+  expect_identical(extract_hash_from_object(""), "2f88e1")
+
+})
+
+test_that("extract_hash_from_object s4 MariaDBResult work", {
+  if (requireNamespace("RMariaDB", quietly = TRUE)) {
+    # create S4-Object for testing
+    obj <- new("MariaDBResult")
+    obj@sql <- "Select * from my_table"
+    expect_identical(extract_hash_from_object(obj), "e00bce")
+  }
+})
+
+test_that("extract_hash_from_object s4 PqResult work", {
+  if (requireNamespace("RPostgres", quietly = TRUE)) {
+    # create S4-Object for testing
+    obj <- new("PqResult")
+    obj@sql <- "Select * from my_table"
+    expect_identical(extract_hash_from_object(obj), "e00bce")
+  }
+})
+
+test_that("extract_hash_from_object s4 SQLiteResult work", {
+  if (requireNamespace("RSQLite", quietly = TRUE)) {
+    # create S4-Object for testing
+    obj <- new("SQLiteResult")
+    obj@sql <- "Select * from my_table"
+    expect_identical(extract_hash_from_object(obj), "e00bce")
+  }
+})
+
+test_that("extract_hash_from_object s4 OdbcResult work", {
+  if (requireNamespace("odbc", quietly = TRUE)) {
```

Comment:
I was getting an error locally about the odbc class not being available. I've added in these `if (requireNamespace(...` bits to make sure the namespaces are available for the tests (but to not fail if the package is not available, which is required on CRAN). I added those up to the other tests for the other packages too (I suspect those namespaces happened to be attached in other tests so they weren't failing locally where all of the packages are available, but CRAN requires that if those packages happen to not be available that the test does not fail so better to pre-guard ourselves here)